sdk/tables/Azure.Data.Tables/src/TableServiceClient.cs

```diff
@@ -911,6 +911,10 @@ public virtual Uri GenerateSasUri(TableAccountSasPermissions permissions, TableA
             TableAccountSasBuilder builder)
         {
             Argument.AssertNotNull(builder, nameof(builder));
+            if (SharedKeyCredential == null)
+            {
+                throw new InvalidOperationException($"{nameof(GenerateSasUri)} requires a credential other than {nameof(TokenCredential)} in order to sign the SAS token.");
```

Comment:
```suggestion
                throw new InvalidOperationException($"{nameof(GenerateSasUri)} requires a credential that is not a {nameof(TokenCredential)} type in order to sign the SAS token.");
```

Reply:
I'm not in love with the suggestion, honestly... but it took me a couple of reads of the original to understand that you're looking for a shared key-based credential.

Reply:
Yeah, I am not in love with what I originally wrote either. I wanted to be less specific than Shared Key, since Connection String auth also works. Although the shared key is also present there, it's not explicitly a SharedKeyCredential.

What I was trying to convey with the current message is roughly - "this will fail if you used the TokenCredential ctor"

Reply:
What do you think of this?

  throw new InvalidOperationException($"{nameof(GenerateSasUri)} requires that this client be constructed with a credential type other than {nameof(TokenCredential)} in order to sign the SAS token.");

Reply:
Nice.... I like that!